
Use the thread-safe LDAP library by default. This also prevents a conflict
with the illumos libldap in /usr/lib.

diff -wpruN --no-dereference '--exclude=*.orig' a~/ext/ldap/config.m4 a/ext/ldap/config.m4
--- a~/ext/ldap/config.m4	1970-01-01 00:00:00
+++ a/ext/ldap/config.m4	1970-01-01 00:00:00
@@ -74,10 +74,10 @@ if test "$PHP_LDAP" != "no"; then
 
   if test -f $LDAP_LIBDIR/liblber.a || test -f $LDAP_LIBDIR/liblber.$SHLIB_SUFFIX_NAME || test -f $LDAP_LIBDIR/$MACHINE_INCLUDES/liblber.a || test -f $LDAP_LIBDIR/$MACHINE_INCLUDES/liblber.$SHLIB_SUFFIX_NAME; then
     PHP_ADD_LIBRARY_WITH_PATH([lber], [$LDAP_LIBDIR], [LDAP_SHARED_LIBADD])
-    PHP_ADD_LIBRARY_WITH_PATH([ldap], [$LDAP_LIBDIR], [LDAP_SHARED_LIBADD])
+    PHP_ADD_LIBRARY_WITH_PATH([ldap_r], [$LDAP_LIBDIR], [LDAP_SHARED_LIBADD])
 
   elif test -f $LDAP_LIBDIR/libldap.$SHLIB_SUFFIX_NAME || test -f $LDAP_LIBDIR/libldap.$SHLIB_SUFFIX_NAME.3 || test -f $LDAP_LIBDIR/$MACHINE_INCLUDES/libldap.$SHLIB_SUFFIX_NAME || test -f $LDAP_LIBDIR/$MACHINE_INCLUDES/libldap.$SHLIB_SUFFIX_NAME.3 || test -f $LDAP_LIBDIR/libldap.3.dylib; then
-    PHP_ADD_LIBRARY_WITH_PATH([ldap], [$LDAP_LIBDIR], [LDAP_SHARED_LIBADD])
+    PHP_ADD_LIBRARY_WITH_PATH([ldap_r], [$LDAP_LIBDIR], [LDAP_SHARED_LIBADD])
 
   elif test -f $LDAP_LIBDIR/libclntsh.$SHLIB_SUFFIX_NAME.12.1 || test -f $LDAP_LIBDIR/$MACHINE_INCLUDES/libclntsh.$SHLIB_SUFFIX_NAME.12.1; then
     PHP_ADD_LIBRARY_WITH_PATH([clntsh], [$LDAP_LIBDIR], [LDAP_SHARED_LIBADD])
